gems/aws-sdk-core/lib/aws-sdk-core/stubbing/stub_data.rb

```diff
@@ -13,12 +13,23 @@ def initialize(operation)
       def stub(data = {})
         stub = EmptyStub.new(@rules).stub
         remove_paging_tokens(stub)
+        remove_checksums(stub)
         apply_data(data, stub)
         stub
       end
 
       private
 
+      def remove_checksums(stub)
+        if @rules && @rules.shape.is_a?(Seahorse::Model::Shapes::StructureShape)
+          @rules.shape.members.each do |key, member|
+            if member.location == 'header' && member.location_name.start_with?('x-amz-checksum-')
```

Comment:
I think we should check for the trait, right? Will checksums always be headers, anywhere, and in the future?

Reply:
The checksums can be on a response without the trait (see s3 head_object).  They MUST be headers and they MUST start with 'x-amz-checksums-`